src/components/todolist/todolist.tsx

```diff
@@ -0,0 +1,178 @@
+// import { TodoInfo } from '../todoinfo/todoinfo';
+import classNames from 'classnames';
+import { useTodos } from '../../context/todoProvider';
+import { deleteTodo } from '../../api/todos';
+import { ErrorType } from '../../types/Error';
+
+export const TodoList = () => {
+  const {
+    visibleTasks, setError, tempTodo, todos,
+    setTodos, deletingTask, setDeletingTask,
+  } = useTodos();
+
+  const handleDeleteClick = (id: number) => {
+    const deletingId = [...deletingTask, id];
+
+    setDeletingTask(deletingId);
```

Comment:
if you have a queue todos for deleting, and api has only delete 1 todo, maybe better to map todos, do deleteOne(id) and await Promise.allSettled, for be confident, that we finished process
Make 2 function, one for delete 1 todo, second for delete all completed

Reply:
it depends on author imagination rather it's an error. I think that this solution works and while it works and is not a problematic thing it should be accepted. Everyone has it's own logic